transform/flatten_mangler.go

```diff
@@ -271,3 +288,41 @@ func getUnderlyingKindType(t reflect.Type) (reflect.Kind, reflect.Type) {
 	}
 	return k, t
 }
+
+func stripPtrs(val reflect.Value) reflect.Value {
+	for val.IsValid() {
+		switch val.Kind() {
+		case reflect.Ptr, reflect.Interface:
+			val = val.Elem()
+		default:
+			return val
+		}
+	}
+	return val
+}
+
+// GetField takes in a reflect.Value and returns the value of the field at the
+// given field path or the zero value if it's not populated
+func GetField(v reflect.Value, t reflect.Type, fieldPath string) interface{} {
+	fields := strings.Split(fieldPath, ",")
+	for _, fname := range fields {
+		v = stripPtrs(v)
+		// if the struct isn't populated, return the zero value
+		if !v.IsValid() {
+			// ignore the kind and get the concrete type
+			_, t = getUnderlyingKindType(t)
+			return reflect.Zero(t).Interface()
```

Comment:
Hmm, checking the [docs on `reflect.Zero`](https://golang.org/pkg/reflect/#Zero), the returned value is neither addressable nor settable, so it wouldn't be useful for flag arguments which need to set the value (by calling the `.Set()` method for those that implement the [`flag.Var` interface](https://golang.org/pkg/flag/#Value))

This isn't as big a problem for scalar-types like `int`, but I'd rather not have a special-case for those if we can avoid it. :)

Most other places in dials we use `reflect.New(t).Elem()` to create new objects of the appropriate type, let's do that here as well.

Another issue is that as-is this will return either the concrete-type that an intermediate field would point to or an intermediate field itself. Either way, it's not the type we actually want to return. (I think dropping the call to `getUnderlyingKindType()` is appropriate here)

Reply:
Good catch, `reflect.New(t).Elem()` would be better. 

As discussed on slack, since `Pointerify` makes everything into pointers, not calling `getUnderlyingKindType` would return pointers even for fields that aren't pointers (wow that's a lot of `pointers` in a sentence). So keeping the `getUnderlyingKindType` to get the concrete values and updated the doc string to specify that the function will return concrete values.